### PR TITLE
ch-user-model-3wek9j3: Create a user model for the DB

### PR DESCRIPTION
**Pull request by codenney**
### What does this PR do?
* This PR is to create the user model using the DB design that we have in place.

### Description of the task to be completed?
* Create the user model file in the model folder and import
* Update the index file to create the table
* Update the config file to sync the table with the database

### How should this be manually tested?
* Sending a request to the DB to create a table or add item(s) to it

### Required
* The tests

### CODE REVIEW

File: src/models/user.model.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+module.exports = (sequelize, DataTypes) => {
+    const User = sequelize.define("User", {
+        id: {
+            type: DataTypes.UUID,
+            defaultValue: DataTypes.UUIDV4,
+            primaryKey: true
+            // type: DataTypes.INTEGER,
```
Comment (Endowmissy) [inline src/models/user.model.js]: @Codenney why did you push this?

File: src/models/user.model.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        },
+        first_name: {
+            type: DataTypes.STRING,
+            allowNull: false
+        },
+        last_name: {
+            type: DataTypes.STRING,
+            allowNull: false
+        },
+        email_name: {
```
Comment (Endowmissy) [inline src/models/user.model.js]: @Codenney this should be email not email_name

File: src/models/user.model.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+            type: DataTypes.BOOLEAN,
+            defaultValue: false,
+            allowNull: false,
+        },
+        is_active: {
+            type: DataTypes.BOOLEAN,
+            defaultValue: false,
+            allowNull: false,
+        },
+        role: {
```
Comment (Endowmissy) [inline src/models/user.model.js:41]: @Codenney this should have a default value

File: src/models/user.model.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            type: DataTypes.ENUM('Admin', 'User'),
```
Comment (Endowmissy) [inline src/models/user.model.js:40]: @Codenney this should have a default value

File: src/models/user.model.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            type: DataTypes.STRING,
+            allowNull: false
+        },
+        email_name: {
+            type: DataTypes.STRING,
+            allowNull: false,
+            unique: true        },
+        password: {
+            type: DataTypes.STRING,
+            allowNull: false
```
Comment (kelue) [inline src/models/user.model.js]: password should have unique constraint

File: src/models/user.model.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        },
+        phone_number: {
+            type: DataTypes.STRING,
+            allowNull: false
```
Comment (kelue) [inline src/models/user.model.js]: phone number should have unique constraints too

**Review by kelue — CHANGES_REQUESTED** [review]

File: database/config.js [hunk 7]
```diff
@@ -1,21 +1,37 @@
...
-    console.error('unable to connect to database');
+    logger.error('unable to connect to database');
   }
 };
 
 databaseConnection();
 
+// Sync all models with the Database
+const syncDB = (db) => {
+  // force: do not drop the table if it already exist but add to it
```
Comment (Endowmissy) [inline database/config.js]: @Codenney why do we need this?

**Pull request closed without merging**